src/content/resources/glossary.md

```diff
@@ -279,3 +279,49 @@ If not, the pattern _refutes_, or denies, the match.
 Refutable patterns appear in [_matching contexts_][].
 
 [_matching contexts_]: /language/patterns#matching
+
+## Subclass
+
+A _subclass_ is a class that inherits the implementation of another class by using the
+[`extends`](/language/extend) keyword, or by [mixin application](#mixin-application).
+
+```dart
+class A extends B {} // A is a subclass of B; B is the superclass of A. 
+
+class B1 with M {} // B1 is a subclass of M.
+```
+
+Subclassing is fundamentally about code reuse.
+
+A subclass relation also implies an associated [subtype](#subtype) relation.
+For example, the declaration `class Foo` implicitly defines
+an associated type `Foo` which instances of the class `Foo` inhabit.
+
+The Dart documentation occasionally uses "subtype"
+to describe subclass relations for this reason.
+However, the converse is not true: not all subtypes are subclasses.
+See the [subtype](#subtype) entry for more information.
+
+## Subtype
+
+A _subtype_ relation is when a type is fully substitutable for another type.
+A subtype supports all of the operations of its supertype
+(and possibly some extra operations).
+In practice, this means that the value of a subtype can be assigned
+to any location expecting the supertype,
+and all of the methods of the supertype are available on the subtype.
+
+Some subtype relations are based solely on the structure of the type,
+like with nullable types (for example, `int` is a subtype of `int?`)
+and function types
+(for example, `String Function()` is a subtype of `void Function()`).
```

Comment:
Two questions, not sure if they're related but trying to understand:

1. Does the fact that `String` is a subtype of `void` fall under "based on the structure of the type"?

2. Could "based solely on the structure of the type" be reasonably extrapolated to include "based on the structure of the _function_ type"

Reply:
This is definitely just nitpicking on my part; I'm mostly cautioning against saying "solely". Practical examples of subtyping typically come from a mix of structural and non-structural rules.

Given any type `T`, we know that `T <: T?`. (`<:` means "is a subtype of".) I think it's fair to say this is based solely on the structure of the type, and no preconditions have to be satisfied.

On the other hand, if we're given types `S` and `T`, in order to claim that `S Function()` is a subtype of `T Function()`, there is a condition we have to satisfy: `S` must be a subtype of `T`. The structure of the type is what lets us go from `S <: T` to `S Function() <: T Function()`, but the requirement `S <: T` itself might be met by a non-structural relation. For example, `int Function()` is a subtype of `num Function()`, in part because of the non-structural subtype relation between `int` and `num`.

To answer your particular questions:
1. Yes, it's kind of a trivial structure, but it comes from the definition of `void`. `void` is a top type, so `T <: void` for *any* `T`.
2. Yes, that extrapolation is reasonable. I'm just cautioning against the word "solely" because many concrete examples of function subtyping will be based on a mix of the structural rules for function subtyping and non-structural relations between the subparts of the function types.

Reply:
Wow ok, thanks for the explanation! Interesting stuff. I did not remove "solely" from the last round of edits so I'll do that now.

Reply:
This isn't necessary for the PR, so no need to answer promptly, but I think I'm confused about structural vs. non-structural type relationships.

You mentioned "the non-structural subtype relation between `int` and `num`".  I guess I was imagining the "structure" of types relations kind of like this [top-and-bottom graph](https://dart.dev/null-safety/understanding-null-safety#top-and-bottom) (literally, some kind of structure), and figured `num` -> `int` fell into that category. 

If that's not what a structural type relation is, what's the difference between structural and non-structural?

---

Also...

> (<: means "is a subtype of".)

Now I can finally understand the type schema section of all the language specs 😄 thank you!

Reply:
"Structural" in this context generally means "based on the internal structure of the types themselves", not the type hierarchy we organize types into. For example, `T?` has some internal structure: there is an underlying type `T`, which is suffixed with `?` to indicate nullability. A function type has structure: the type `void Function(int)` is composed from the return type `void` and the parameter type `int`. Such types generally have subtyping rules which are expressed in terms of their substructures, and the programmer doesn't opt into or out of these relations. These are baked into the type system itself, not declared.

This is usually contrasted with nominal subtyping, which is based on declarations/names the programmer provides. `class int` on its own has no special structure or significance to the type system. It's a subtype of `num` because of the `extends num` clause in the declaration.

Reply:
Oh, got it that's very clear now! Thanks again